src/platforms/ctxlink/WiFi_Server.c

```diff
@@ -641,6 +655,130 @@ bool is_ip_address_assigned(void)
 	return res;
 }
 
+//
+// Format the current connection data for display to user
+//
+//	SSID = 'name'
+//	RSSI = xx
+//	ip = xxx.xxx.xxx.xxx
+//
+void wifi_get_ip_address(char *buffer, uint32_t size)
+{
+	char local_buffer[64] = {0};
+	memset(buffer, 0x00, size - 1);
+	if (g_wifi_connected) {
+		snprintf(local_buffer, sizeof(local_buffer), "SSID = %s\n", conn_info.acSSID);
+		strncpy(buffer, local_buffer, size);
+		snprintf(local_buffer, sizeof(local_buffer), "RSSI = %d\n", conn_info.s8RSSI);
+		strncat(buffer, local_buffer, size);
+		snprintf(local_buffer, sizeof(local_buffer), "IP = %d.%d.%d.%d\n", conn_info.au8IPAddr[0],
+			conn_info.au8IPAddr[1], conn_info.au8IPAddr[2], conn_info.au8IPAddr[3]);
+		strncat(buffer, local_buffer, size);
+	} else {
+		memcpy(buffer, "Not connected\n", strlen("Not connected\n"));
+	}
+}
+
+//
+// Wait for app_state to spin with timeout
+//
+void app_task_wait_spin(void)
+{
+	uint32_t wait_timeout = 2000U;
+	while (1) {
+		platform_tasks();
+		if (app_state == app_state_spin)
+			break;
+		platform_delay(1);
+		if (wait_timeout-- == 0)
+			break;
+	}
+}
+
+//
+// Using the passed arguments, attempt to connect to a Wi-Fi AP
+//
+void wifi_connect(int argc, const char **argv, char *buffer, uint32_t size)
```

Comment:
As you check `argc` is at least 1 in platform_wifi_state(), please switch the type of the parameter here to `size_t` and in the loop below to avoid potential boundary condition problems in the `loop < argc` check. In practice this should never be a problem, but C doesn't make it a guarantee unless you're using unsigned types due to overflow UB on signed ones.

Reply:
Fixed.